packages/uniforms-mui/src/BoolField.tsx

```diff
@@ -0,0 +1,72 @@
+import Checkbox, { CheckboxProps } from '@mui/material/Checkbox';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import FormGroup from '@mui/material/FormGroup';
+import FormLabel from '@mui/material/FormLabel';
+import Switch, { SwitchProps } from '@mui/material/Switch';
+import omit from 'lodash/omit';
+import React, { Ref } from 'react';
+import { FieldProps, connectField, filterDOMProps } from 'uniforms';
+
+import wrapField from './wrapField';
+
+export type BoolFieldProps = FieldProps<
+  boolean,
+  CheckboxProps | SwitchProps,
+  {
+    appearance?: 'checkbox' | 'switch';
+    fullWidth?: boolean;
+    helperText?: string;
+    legend?: string;
+    margin?: 'dense' | 'normal' | 'none';
+    transform?: (label: string) => string;
+  }
+>;
+
+function Bool(props: BoolFieldProps) {
+  const {
+    appearance,
+    disabled,
+    inputRef,
+    label,
+    legend,
+    name,
+    onChange,
+    readOnly,
+    transform,
+    value,
+  } = props;
+  const SelectionControl =
+    appearance === 'checkbox' || appearance === undefined ? Checkbox : Switch;
+
+  return wrapField(
+    { fullWidth: true, margin: 'dense', ...props, component: 'fieldset' },
```

Comment:
While doing this, please do not remove possibility to set text field size by passing either `margin` or `size` props (e.g `margin="dense"` is pretty much similar to `size="small"`). In current material-ui 4 theme only `margin` prop can be passed to AutoField

Reply:
We definitely won't do that! I just want to make it "as default as possible".

Also, can you report this issue separately? I think it'd be something more people could benefit from.

Reply:
I think broken size was already mentioned and resolved here https://github.com/vazco/uniforms/issues/894
I've actually thought supported props are whitelisted, but it's more like unsupported are blacklisted
So yeah, it might be good idea to keep everything as stock as possible not to complicate customization

Reply:
@radekmie I think it makes more sense for it to be less opinionated because the user will have full control over the default theme props anyway via MUI theming. Such an approach would basically make #993 obsolete. I'm all for changing it within this PR.

Reply:
Then let's do it. The less we do, the better.

Reply:
@radekmie `fieldset` is not required to properly render checkboxes so it's mostly up to us whether it makes sense here. Since we only have a singular checkbox and it has a label next to it anyway. Removed these props for now.